tests/Exception/LimitExceededTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RateLimit\Tests\Exception;
+
+use PHPUnit\Framework\TestCase;
+use RateLimit\Exception\LimitExceeded;
+use RateLimit\Rate;
+
+/**
+ * @covers \RateLimit\Exception\LimitExceeded
+ */
+final class LimitExceededTest extends TestCase
+{
+    public function testForReturnsException(): void
+    {
+        $identifier = 'foo';
+        $rate = Rate::custom(
+            100,
+            3600
+        );
+
+        $exception = LimitExceeded::for(
+            $identifier,
+            $rate
+        );
+
+        $expected = sprintf(
```

Comment:
@ManInTheBox

Does this make more sense to you?

Reply:
Thanks, looking good except that it can sound weird if you use the factory method e.g. [`Rate::perHour()`](https://github.com/nikolaposa/rate-limit/blob/b5e9097f5d1406dc708bd207ad69db6c2b4f27bd/src/Rate.php#L36) and you get an error message telling you the limit was exceeded for 3600 seconds. Maybe no metion of the interval would be just a good idea.

Reply:
@ManInTheBox

Adjusted!

Reply:
One of the reasons I ended up removing both while I was initially working on this particular exception. :) Perhaps some sophisticated time formatter would do, but I really I would not put much effort into the message itself since all useful information can be obtained from captured variables.

Reply:
So here are my thoughts about this one... If you compare these messages:
a) `Limit has been exceeded for identifier "10.10.10.10".` – sensible, except that it is missing rate configuration information
b) `Limit of 100 operation(s) has been exceeded for identifier "10.10.10.10".` – vague, as it includes `operations` only 
c) `Limit of 100/60 has been exceeded for identifier "10.10.10.10".` – doesn't look pretty, because `interval` part is always seconds
d) `Limit of 100 operations per minute has been exceeded for "10.10.10.10".` – ✨ 

But the only way to achieve d) would be to preserve `second|minute|hour|day` in the `Rate` itself after some of its named constructors is invoked, so that it can be used for formatting or similar purposes. Is it worth doing something like that just because of this exception message? I'm not sure. I personally don't care much about the message, but the ability to easily obtain all the necessary information from the exception, which is already enabled via `getIdentifier()` and `getRate()`.

Thoughts?

Reply:
Perhaps `a` is good enough.

Reply:
Adjusted!

Reply:
Thanks 👍